tests/modules/submissions/resources/test_permissions.py

```diff
@@ -0,0 +1,125 @@
+):
+    # Clone as the regular user and then try to reread as both regular and readonly user,
+    # read by regular user should succeed, read by readonly user should be blocked
+
+    clone = clone_submission(
+        flask_app_client,
+        regular_user,
+        test_clone_submission_data['submission_uuid'],
+        later_usage=True,
+    )
+
+    try:
+        with flask_app_client.login(
+            regular_user,
+            auth_scopes=(
+                'submissions:read',
+                'assets:read',
+            ),
+        ):
+            submission_response = flask_app_client.get(
+                '/api/v1/submissions/%s' % test_clone_submission_data['submission_uuid']
+            )
+            asset_response = flask_app_client.get(
+                '/api/v1/assets/%s' % test_clone_submission_data['asset_uuids'][0]
+            )
+            assert submission_response.status_code == 200
+            assert asset_response.status_code == 200
+
+        with flask_app_client.login(
+            readonly_user,
+            auth_scopes=(
+                'submissions:read',
+                'assets_read',
+            ),
+        ):
+            submission_response = flask_app_client.get(
+                '/api/v1/submissions/%s' % test_clone_submission_data['submission_uuid']
+            )
+            asset_response = flask_app_client.get(
+                '/api/v1/assets/%s' % test_clone_submission_data['asset_uuids'][0]
+            )
+            assert submission_response.status_code == 403
+            assert asset_response.status_code == 401
+
+        # and as no user, removed for now as the
+        # @api.login_required(oauth_scopes=['submissions:read']) has been added back to
+        # class SubmissionByID and the anonymous user is blocked by that
+        # submission_response = flask_app_client.get(
+        #        '/api/v1/submissions/%s' % test_clone_submission_uuid
+        #    )
+        # assert submission_response.status_code == 403
+
+    except Exception as ex:
+        raise ex
+    finally:
+        clone.cleanup()
+
+
+def test_create_patch_submission(flask_app_client, regular_user, readonly_user, db):
+    # pylint: disable=invalid-name
+    submission_guid = None
+
+    try:
+        from app.modules.submissions.models import Submission, SubmissionMajorType
+
+        test_major_type = SubmissionMajorType.test
+
+        with flask_app_client.login(regular_user, auth_scopes=('submissions:write',)):
+            create_response = flask_app_client.post(
+                '/api/v1/submissions/',
+                data=json.dumps(
+                    {
+                        'major_type': test_major_type,
+                        'description': 'This is a test submission, please ignore',
+                    }
+                ),
+            )
+
+        # @todo patch via regular user also fails but for a different reason, not permission
```

Comment:
This is something that I don't understand. I would have thought a patch by regular user shoudl work but it doesn't Likewise delete fails with error 500

Reply:
Will make a ticket for this so we can dig into it on the Wed meeting, I'd like to learn about what is going on.